assets/nginx.mtail

```diff
@@ -22,44 +26,68 @@ counter nginx_response_size_bytes_sum by vhost, method, code, cache_status
 /"(?P<upstream_cache_status>[[:print:]]+)?"/ +
 /$/ {
   nginx_requests_total[$vhost][tolower($request_method)][$status][tolower($upstream_cache_status)]++
+
   nginx_response_size_bytes_sum[$vhost][tolower($request_method)][$status][tolower($upstream_cache_status)] += $bytes_sent
 
+  nginx_request_duration_milliseconds_count[$vhost][tolower($request_method)][$status][tolower($upstream_cache_status)]++
+  nginx_request_duration_milliseconds_sum[$vhost][tolower($request_method)][$status] += $request_seconds * 1000 + $request_milliseconds < 100
+
   $request_seconds * 1000 + $request_milliseconds < 100 {
-    nginx_request_duration_milliseconds_sum["100"][$vhost][tolower($request_method)][$status] += $request_seconds * 1000 + $request_milliseconds
+    nginx_request_duration_milliseconds_bucket["100"][$vhost][tolower($request_method)][$status]++
+  } else {
+    nginx_request_duration_milliseconds_bucket["100"][$vhost][tolower($request_method)][$status] += 0
   }
 
   $request_seconds * 1000 + $request_milliseconds < 200 {
-    nginx_request_duration_milliseconds_sum["200"][$vhost][tolower($request_method)][$status] += $request_seconds * 1000 + $request_milliseconds
+    nginx_request_duration_milliseconds_bucket["200"][$vhost][tolower($request_method)][$status]++
+  } else {
+    nginx_request_duration_milliseconds_bucket["200"][$vhost][tolower($request_method)][$status] += 0
   }
 
   $request_seconds * 1000 + $request_milliseconds < 300 {
-    nginx_request_duration_milliseconds_sum["300"][$vhost][tolower($request_method)][$status] += $request_seconds * 1000 + $request_milliseconds
+    nginx_request_duration_milliseconds_bucket["300"][$vhost][tolower($request_method)][$status]++
+  } else {
+    nginx_request_duration_milliseconds_bucket["300"][$vhost][tolower($request_method)][$status] += 0
   }
 
   $request_seconds * 1000 + $request_milliseconds < 500 {
-    nginx_request_duration_milliseconds_sum["500"][$vhost][tolower($request_method)][$status] += $request_seconds * 1000 + $request_milliseconds
+    nginx_request_duration_milliseconds_bucket["500"][$vhost][tolower($request_method)][$status]++
+  } else {
+    nginx_request_duration_milliseconds_bucket["500"][$vhost][tolower($request_method)][$status] += 0
```

Comment:
What is the purpose of this `else` block?

Reply:
So, without it the bucket which do not exist have no value, therefore will not be scraped. This ensures that the value is `0` instead of null.